subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/BuildScriptCompileAvoidanceIntegrationTest.kt

```diff
@@ -584,9 +585,20 @@ class BuildScriptCompileAvoidanceIntegrationTest : AbstractKotlinIntegrationTest
         configureProject().assertBuildScriptCompilationAvoided()
     }
 
+    private
+    fun withKotlinDslPluginInBuildSrc() {
+        // this is to force buildSrc/build.gradle.kts to be written to test-local buildscript cache
+        // and not to be reused from daemon's cache from other tests when daemon is in use
+        withBuildScriptIn("buildSrc", scriptWithKotlinDslPlugin()).appendText(
+            """
+                println("$cacheBuster")
```

Comment:
Maybe just do `val randomId = "$cacheBuster"` to not have random Strings printed into the log without context